orange_cb_recsys/recsys/ranking_algorithms/page_rank.py

```diff
@@ -11,85 +12,311 @@
+        """
+        Setter for the boolean attribute remove_properties
+        True if you want to remove property nodes, False otherwise
+        """
+        self.__remove_properties = remove_properties
 
     @abstractmethod
     def predict(self, ratings: pd.DataFrame, recs_number: int, items_directory: str,
                 candidate_item_id_list: List = None):
+        """
+        Abstract method that should encapsulate the computation process for the PageRank
+        """
         raise NotImplementedError
 
-    def clean_rank(self, rank: Dict, user_id: str,
-                   remove_from_nodes: bool = True,
-                   remove_profile: bool = True,
-                   remove_properties: bool = True) -> Dict:
-        extracted_profile = self.extract_profile(user_id)
+    def clean_rank(self, rank: Dict, graph: FullGraph, user_ratings: pd.DataFrame, user_id: str = None) -> Dict:
+        """
+        Cleans a rank from all the nodes that are not requested. It's possible to remove user nodes,
+        property nodes and item nodes, the latter if they are already in the user profile. This produces a filtered
+        ranking with only the desired nodes inside of it. What is filtered depends by the
+        attributes remove_user_nodes, remove_items_in_profile and remove_properties
+
+        Args:
+            rank (dict): dictionary representing the ranking (keys are nodes and values are their ranked score)
+            graph (FullGraph): graph from which the user profile will be extracted
+            user_ratings (pd.Dataframe): dataframe containing the ratings of a user, used in the extraction of the
+                user profile
+            user_id (str): id of the user used to extract his profile (if None the profile will be empty)
+
+        Returns:
+            new_rank (dict): dictionary representing the filtered ranking
+        """
+        if user_id is not None:
+            extracted_profile = self.extract_profile(user_id, graph, user_ratings)
+        else:
+            extracted_profile = {}
+
         new_rank = {k: rank[k] for k in rank.keys()}
         for k in rank.keys():
-            if remove_from_nodes and self.__fullgraph.is_user_node(k):
+            if self.__remove_user_nodes and graph.is_user_node(k):
                 new_rank.pop(k)
-            if remove_profile and self.__fullgraph.is_item_node(k) and k in extracted_profile.keys():
+            elif self.__remove_items_in_profile and graph.is_item_node(k) and k in extracted_profile.keys():
                 new_rank.pop(k)
-            if remove_properties and self.__fullgraph.is_property_node(k):
+            elif self.__remove_properties and graph.is_property_node(k):
                 new_rank.pop(k)
+
         return new_rank
 
-    def extract_profile(self, user_id: str) -> Dict:
-        succ = self.__fullgraph.get_successors(user_id)
+    @staticmethod
+    def extract_profile(user_id: str, graph, user_ratings: pd.DataFrame) -> Dict:
```

Comment:
- perché si dà la possibilità di estrarre il profilo da un grafo generico e non quello passato nel costruttore? (l'attributo self.__fullgraph per capirci)
- Inoltre non serve più il rating_frame, implementazione vecchia non effettuava una conversione in un grafo esatta e si portava dietro il rating frame con il quale creava il grafo, adesso la conversione in grafo è completa e puoi accedere agli item votati da un utente solo e soltanto coi metodi del grafo